tests/virt_autotest/switch_version_and_reload_needle.pm

```diff
@@ -0,0 +1,29 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This test does not change the VERSION and only reload needles.
+#          It needs to be wrapped in test because
+#          set_var with reload_needles option needs to be run in test files rather than main.pm,
+#          It does not change VERSION together with reload_needles, because
+#          VERSION needs to be changed in main.pm, so as to load different tests accordingly
+# Maintainer: xlai@suse.com
+
+
+use strict;
+use warnings;
+use base "virt_autotest_base";
+use testapi;
+
+sub run {
+    #VERSION needs to be changed in main.pm, so as to load different tests accordingly
+    set_var('VERSION', get_var('VERSION'), reload_needles => 1);
```

Comment:
This looks a bit strange obviously because you are setting VERSION to VERSION. Maybe it would work better to set the variables as you do in the main.pm here instead, call the test module explicitly depending on what it switches to. Then you should also not need a `save_vars()` call in the main.pm (if that is necessary at all).

Reply:
Please see comment above.